### ISSUE TITLE
Incorrect memory size reported for types using null pointer optimization

### ISSUE THREAD

**Issue by radioactiveray** [rust-lang/rust-analyzer#22539]
Memory layout size for structures and type aliases using null pointer optimization is provided wrongfully:

<img width="421" height="358" alt="Image" src="https://github.com/user-attachments/assets/e49e470d-a288-4f76-9e98-cc9bbe5553cf" />

<img width="481" height="312" alt="Image" src="https://github.com/user-attachments/assets/eed219d0-3d7d-49c7-a24d-919f40c3dafe" />

The size in both cases is reported as 16 bytes even though null pointer optimization guarantees 8 byte layout and through the ```assert!``` it gets clear that this is not a ```rustc``` bug, but rather a ```rust-analyzer``` bug (at least I assume so).

**rust-analyzer version**: rust-analyzer version: 0.3.2921-standalone (c5d30e2331 2026-05-31)

**rustc version**: rustc 1.96.0 (ac68faa20 2026-05-25)

**editor or extension**: VSCode version: 1.123.0

**code snippet to reproduce**:
```rust
#[repr(transparent)]
pub struct ExampleA(Option<NonNull<()>>);

// does no fail, as expected
const _: () = assert!(size_of::<ExampleA>() == 8);

#[repr(transparent)]
pub struct ExampleB(NonNull<()>);

// does not fail, as expected
const _: () = assert!(size_of::<ExampleB>() == 8);

pub type ExampleC = Option<ExampleB>;

// does not fail, as expected
const _: () = assert!(size_of::<ExampleC>() == 8);

```
